src/contexts/auth-context.js

```diff
@@ -0,0 +1,91 @@
+import { createContext, useContext, useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { loginServices, signupSevices } from '../Services/Services';
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const localStorageToken = JSON.parse(localStorage.getItem('login'));
+  const [token, setToken] = useState(localStorageToken?.token);
+  const localStorageUser = JSON.parse(localStorage.getItem('user'));
+  const [user, setUser] = useState(localStorageUser?.user);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    let id;
+    if (token) {
+      id = setTimeout(() => {
+        navigate('/video');
+      });
+    }
+    return () => clearTimeout(id);
+  }, [token]);
+
+  const loginHandler = async (e, setLogin, login) => {
+    e.preventDefault();
+    try {
+      let resp;
+      if (e.target.innerText === 'Sign In as Guest') {
```

Comment:
could do in a better way! Kinda hardcoding

Reply:
Sure I'll Explore.